os/kernel/task/task_exit.c

```diff
@@ -127,9 +127,18 @@ int task_exit(void)
 	FAR struct tcb_s *dtcb = this_task();
 	FAR struct tcb_s *rtcb;
 	int ret;
+	struct mm_heap_s *heap = mm_get_heap_info();
 
 	trace_begin(TTRACE_TAG_TASK, "task_exit");
 
+	/* Since context switching should not occur during task_exit,
+	 * in order to release the heap used by dtcb, it is necessary
+	 * to check whether heap->mm_semaphore is acquired before executing exit.
+	 */
```

Comment:
Need to add additional description like:
Otherwise, the critical section of mm operations might not be protected because the exiting task's pid is inevitably changed to one of the head of readytorun list. Unfortunatetly, if the holder of heap semaphore is the head of readytorun, then the thread of task_exit function re-enters and breaks the critical section.